src/python/dxpy/dx_extract_utils/ExpressionInputsValidator.py

```diff
@@ -0,0 +1,159 @@
+class ExpressionInputsValidator:
+    """InputsValidator class for extract_assay expresion. Checks for invalid input combinations"""
+
+    def __init__(self, args, error_handler=print):
+        self.args = args
+        self.error_handler = error_handler
+
+    def validate(self):
+        self.end_arguments_check("list_assays")
+        self.end_arguments_check("additional_fields_help")
+
+        invalid_combination_infos = {
+            "json_help_unique": {
+                "invalid_combination": self.args["json_help"]
+                and any(
+                    [
+                        self.args["assay_name"],
+                        self.args["sql"],
+                        self.args["additional_fields"],
+                        self.args["expression_matrix"],
+                        self.args["output"],
+                    ]
+                ),
+                "error_message": '"--json-help" cannot be passed with any of "--assay-name", "--sql", "--additional-fields", "--expression-matrix", or "--output"',
+            },
+            "retrieve_expression_json": {
+                "invalid_combination": self.args["retrieve_expression"] and not any(
+            [self.args["input_json"], self.args["json_help"]]),
+                "error_message": 'The flag "--retrieve_expression" must be followed by a json input or json help.',
+            },
+            "empty_json": {
+                "invalid_combination": self.args["input_json"] == "{}",
+                "error_message": 'JSON for "--retrieve-expression" does not contain valid filter information.',
+            },
+        }
+        self.invalid_combinations_check(invalid_combination_infos)
+
+    def end_arguments_check(self, end_arg):
+        args_to_check = [
+            "list_assays",
+            "retrieve_expression",
+            "assay_name",
+            "input_json",
+            "input_json_file",
+            "json_help",
+            "sql",
+            "additional_fields",
+            "expression_matrix",
+            "delim",
+            "output",
+        ]
+        args_specific_list = [self.args[e] for e in args_to_check if e != end_arg]
+        invalid_combination = self.args[end_arg] and any(args_specific_list)
+
+        if invalid_combination:
+            self.error_handler(
+                '"--{}" cannot be presented with other options.'.format(
+                    end_arg.replace("_", "-")
+                )
+            )
+
+    def invalid_combinations_check(self, invalid_combinations_info):
```

Comment:
yes but I think it makes sense to leave which kind of checking is being done. I created some more general validation methods and with those it might make sense to leave it as is. Let me know your thoughts.